core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -821,11 +858,60 @@ object Parser {
 
   /** Represents where a failure occurred and all the expectations that were broken
     */
-  final case class Error(failedAtOffset: Int, expected: NonEmptyList[Expectation]) {
+  final case class Error(input: String, failedAtOffset: Int, expected: NonEmptyList[Expectation]) {
```

Comment:
this is binary and source incompatible and we have been very reluctant to break people.

I might suggest we add a new case class:

```scala
final case class InputAndError(input: String, error: Error)
```
add `def withInput(input: String): InputAndError` to Error and then make the Show instance on that.

How does that sound?

It is probably a design flaw we didn't include the input with the Error, but I really don't want to break compatibility.

Maybe we can find a source and binary compatible way to add the input (for instance, make Error not a case class, but manually add back all the methods that are generated).

Reply:
I Implemented the second option where I coded manually what the Scala compiler would output for a case class